Manual test is completed successfully.

**Setup**

Patch `conduct_version.py` to fail.

```
Felixs-MacBook-Pro-2:conductr-cli felixsatyaputra$ git diff
diff --git a/conductr_cli/conduct_version.py b/conductr_cli/conduct_version.py
index 79adf20..0a659e7 100644
--- a/conductr_cli/conduct_version.py
+++ b/conductr_cli/conduct_version.py
@@ -11,4 +11,5 @@ def version(args):
     log = logging.getLogger(__name__)
     log.screen(__version__)
     log.screen('Supported API version(s): {}'.format(', '.join(supported_api_versions())))
+    raise "boom"
     return True
```

Install the CLI locally using `pip3 install -e .`

Remove the `~/.conductr` directory.

**Result**

Run `conduct version` and inspect the log directory.

```
Felixs-MacBook-Pro-2:conductr-cli felixsatyaputra$ ll ~/.conductr
ls: /Users/felixsatyaputra/.conductr: No such file or directory

Felixs-MacBook-Pro-2:conductr-cli felixsatyaputra$ conduct version
0.41
Supported API version(s): 1, 2
Error: Encountered unexpected error.
Error: Reason: TypeError exceptions must derive from BaseException
Error: Further information of the error can be found in the error log file: /Users/felixsatyaputra/.conductr/errors.log

Felixs-MacBook-Pro-2:conductr-cli felixsatyaputra$ ll ~/.conductr
total 8
drwxr-xr-x   3 felixsatyaputra  staff   102B  4 Nov 10:53 .
drwxr-xr-x+ 66 felixsatyaputra  staff   2.2K  4 Nov 10:53 ..
-rw-r--r--   1 felixsatyaputra  staff   772B  4 Nov 10:53 errors.log

Felixs-MacBook-Pro-2:conductr-cli felixsatyaputra$ cat ~/.conductr/errors.log
2016-11-04 10:53:04,511: Failure running the following command: ['/usr/local/bin/conduct', 'version']
Traceback (most recent call last):
  File "/Users/felixsatyaputra/workspace/typesafe-fsat/conductr-cli/conductr_cli/main_handler.py", line 9, in run
    result = callback()
  File "/Users/felixsatyaputra/workspace/typesafe-fsat/conductr-cli/conductr_cli/conduct.py", line 6, in main_method
    conduct_main.run()
  File "/Users/felixsatyaputra/workspace/typesafe-fsat/conductr-cli/conductr_cli/conduct_main.py", line 371, in run
    is_completed_without_error = args.func(args)
  File "/Users/felixsatyaputra/workspace/typesafe-fsat/conductr-cli/conductr_cli/conduct_version.py", line 14, in version
    raise "boom"
TypeError: exceptions must derive from BaseException
```

Re-run `conduct version again`, logs are appended as expected (i.e. no error in relation to log directory being present when calling `os.makedirs`)

```
Felixs-MacBook-Pro-2:conductr-cli felixsatyaputra$ conduct version
0.41
Supported API version(s): 1, 2
Error: Encountered unexpected error.
Error: Reason: TypeError exceptions must derive from BaseException
Error: Further information of the error can be found in the error log file: /Users/felixsatyaputra/.conductr/errors.log

Felixs-MacBook-Pro-2:conductr-cli felixsatyaputra$ cat ~/.conductr/errors.log
2016-11-04 10:53:04,511: Failure running the following command: ['/usr/local/bin/conduct', 'version']
Traceback (most recent call last):
  File "/Users/felixsatyaputra/workspace/typesafe-fsat/conductr-cli/conductr_cli/main_handler.py", line 9, in run
    result = callback()
  File "/Users/felixsatyaputra/workspace/typesafe-fsat/conductr-cli/conductr_cli/conduct.py", line 6, in main_method
    conduct_main.run()
  File "/Users/felixsatyaputra/workspace/typesafe-fsat/conductr-cli/conductr_cli/conduct_main.py", line 371, in run
    is_completed_without_error = args.func(args)
  File "/Users/felixsatyaputra/workspace/typesafe-fsat/conductr-cli/conductr_cli/conduct_version.py", line 14, in version
    raise "boom"
TypeError: exceptions must derive from BaseException
2016-11-04 10:54:12,885: Failure running the following command: ['/usr/local/bin/conduct', 'version']
Traceback (most recent call last):
  File "/Users/felixsatyaputra/workspace/typesafe-fsat/conductr-cli/conductr_cli/main_handler.py", line 9, in run
    result = callback()
  File "/Users/felixsatyaputra/workspace/typesafe-fsat/conductr-cli/conductr_cli/conduct.py", line 6, in main_method
    conduct_main.run()
  File "/Users/felixsatyaputra/workspace/typesafe-fsat/conductr-cli/conductr_cli/conduct_main.py", line 371, in run
    is_completed_without_error = args.func(args)
  File "/Users/felixsatyaputra/workspace/typesafe-fsat/conductr-cli/conductr_cli/conduct_version.py", line 14, in version
    raise "boom"
TypeError: exceptions must derive from BaseException
```

--- a/conductr_cli/conduct_dcos.py
+++ b/conductr_cli/conduct_dcos.py
@@ -18,7 +18,7 @@ def setup(args):
     if os.path.exists(dst):
         os.remove(dst)
     else:
-        os.makedirs(os.path.dirname(dst))
+        os.makedirs(os.path.dirname(dst), exist_ok=True)
     os.symlink(src, dst)
     log.screen('The DC/OS CLI is now configured.\n'
                'Prefix \'conduct\' with \'dcos\' when you want to contact ConductR on DC/OS e.g. \'dcos conduct info\'')

--- a/conductr_cli/main_handler.py
+++ b/conductr_cli/main_handler.py
@@ -19,7 +19,7 @@ def run(callback):
 
         # Ensure log dir is present before errors are being logged
         log_dir = os.path.abspath(os.path.join(DEFAULT_ERROR_LOG_FILE, '..'))
-        os.makedirs(log_dir)
+        os.makedirs(log_dir, exist_ok=True)
 
         log = logging.getLogger('conductr_cli.main')
         log.error('Encountered unexpected error.')

--- a/conductr_cli/test/test_conduct_dcos.py
+++ b/conductr_cli/test/test_conduct_dcos.py
@@ -14,6 +14,7 @@ class TestConductDcosCommand(CliTestCase):
     def test_success(self):
         args = MagicMock()
         stdout = MagicMock()
+        makedirs = MagicMock()
         symlink = MagicMock()
 
         logging_setup.configure_logging(args, stdout)
@@ -21,10 +22,43 @@ class TestConductDcosCommand(CliTestCase):
         with patch('shutil.which', lambda x: '/somefile'), \
                 patch('os.path.exists', lambda x: True), \
                 patch('os.remove', lambda x: True), \
+                patch('os.makedirs', makedirs), \
                 patch('os.symlink', symlink):
             result = conduct_dcos.setup(args)
         self.assertTrue(result)
 
+        makedirs.assert_not_called()
+
+        symlink.assert_called_with('/somefile',
+                                   '{}/.dcos/subcommands/conductr/env/bin/dcos-conduct'.format(os.path.expanduser('~')))
+
+        self.assertEqual(
+            strip_margin("""|The DC/OS CLI is now configured.
+                            |Prefix \'conduct\' with \'dcos\' when you want to contact ConductR on DC/OS e.g. \'dcos conduct info\'
+                            |"""),
+            self.output(stdout))
+
+    def test_create_dir(self):
+        args = MagicMock()
+        stdout = MagicMock()
+        remove = MagicMock()
+        makedirs = MagicMock()
+        symlink = MagicMock()
+
+        logging_setup.configure_logging(args, stdout)
+
+        with patch('shutil.which', lambda x: '/somefile'), \
+                patch('os.path.exists', lambda x: False), \
+                patch('os.remove', remove), \
+                patch('os.makedirs', makedirs), \
+                patch('os.symlink', symlink):
+            result = conduct_dcos.setup(args)
+        self.assertTrue(result)
+
+        remove.assert_not_called()
+
+        makedirs.assert_called_with('{}/.dcos/subcommands/conductr/env/bin'.format(os.path.expanduser('~')),
+                                    exist_ok=True)
         symlink.assert_called_with('/somefile',
                                    '{}/.dcos/subcommands/conductr/env/bin/dcos-conduct'.format(os.path.expanduser('~')))
 

--- a/conductr_cli/test/test_main_handler.py
+++ b/conductr_cli/test/test_main_handler.py
@@ -56,7 +56,7 @@ class TestMainHandler(CliTestCase):
                 patch('sys.exit', sys_exit_mock):
             main_handler.run(raise_unhandled_error)
 
-        makedirs_mock.assert_called_with(DEFAULT_CLI_SETTINGS_DIR)
+        makedirs_mock.assert_called_with(DEFAULT_CLI_SETTINGS_DIR, exist_ok=True)
 
         self.assertEqual(get_logger_mock.call_args_list, [
             call('conductr_cli.main'),

